mlflow/store/db_migrations/versions/867495a8f9d4_add_trace_tables.py

```diff
@@ -51,6 +51,7 @@ def upgrade():
             sa.ForeignKey(
                 column=SqlTraceInfo.request_id,
                 name=f"fk_{SqlTraceTag.__tablename__}_request_id",
+                ondelete="CASCADE",
```

Comment:
Does this in-place change work if user already has the table created?

Reply:
Good call, we might need migration.

Reply:
Great point! Indeed I was not sure if the alembic migration works for altering cascading behavior (it may, but it doesn't handle relationship well per my understanding), so short-cut it given that we haven't officially release the trace feature...
But yea for safer approach, let me try if we can actually make it work with the migration script.

Reply:
i wonder if we might need to write a new migration, but i'm not sure

Reply:
Since we're not announcing, maybe we can sneak this in 🤣 It seems so complicated

Reply:
> Since we're not announcing, maybe we can sneak this in 🤣 It seems so complicated

Yeah that was my first impression lol😂 

> there's a pr that tried to fix a similar issue when deleting experiments with datasets (https://github.com/mlflow/mlflow/pull/11695) but it got reverted in https://github.com/mlflow/mlflow/pull/11945

Actually this linked PR seems to some feasible solution. The PR was reverted due to conflict with another revision, but the migration script itself looks fine - and pretty sure there is no conflict now. Let me try this one first!

Reply:
It actually worked, yay! Thanks a lot @daniellok-db for the reference!🙌